lib/metasploit/framework/login_scanner/teamcity.rb

```diff
@@ -209,18 +226,21 @@ def try_login(username, password, public_key, retry_counter = 0)
           # Currently, those building blocks are not available, so this is the approach I have implemented.
           timeout = res.body.match(/login only in (?<timeout>\d+)s/)&.named_captures&.dig('timeout')&.to_i
           if timeout
-            framework_module.print_status "User '#{username}' locked out for #{timeout} seconds. Sleeping, and retrying..."
-            sleep(timeout + 1) # + 1 as TeamCity is off-by-one when reporting the lockout timer.
-            result = try_login(username, password, public_key, retry_counter + 1)
-            return result
+            logger.print_status "#{@host}:#{@port} - User '#{username}:#{password}' locked out for #{timeout} seconds. Sleeping, and retrying..." if logger
+            ::IO.select(nil, nil, nil, timeout + 1)
+            return try_login(username, password, public_key, retry_counter + 1)
           end
 
           return { status: ::Metasploit::Model::Login::Status::INCORRECT, proof: res } if res.body.match?('Incorrect username or password')
 
-          raise DecryptionException, 'The server failed to decrypt the encrypted password' if res.body.match?('DecryptionFailedException')
+          raise DecryptionError, 'The server failed to decrypt the encrypted password' if res.body.match?('DecryptionFailedException')
           raise PublicKeyExpiredError, 'The server public key has expired' if res.body.match?('publicKeyExpired')
 
-          { status: :success, proof: res }
+          successful_login_body = %r{^<response><redirect>/favorite/projects</redirect><user id="[0-9]+" /><errors /></response>$}
```

Comment:
I'm worried about this being brittle 🤔 

Other possibilities:
Maybe just status code 200? Or is it a 301 or similar? Potentially just `<errors />`? Something else? 👀